fqdn: Add a nil check for security id lookup

aditighag opened the pull request:
The security id lookup could return nil if the cache
isn't initialized during endpoints restore time,
resulting in a crash.

Signed-off-by: Aditi Ghag <aditi@cilium.io>

Fixes: #13881

joestringer commented on daemon/cmd/fqdn.go:
```diff
@@ -470,13 +470,15 @@ func (d *Daemon) notifyOnDNSMsg(lookupTime time.Time, ep *endpoint.Endpoint, epI
-					LabelsSHA256: secID.GetLabelsSHA256(),
-					Identity:     uint64(serverSecID.ID.Uint32()),
-					Port:         uint16(serverPort),
+				if secID != nil {
```
I presume this means that we will still send a log message, but it will be missing destination endpoint information.

I wonder whether instead we should just try to do a check earlier in the function for whether the identity cache is initialized, and drop the entire notification if the daemon is not ready?

@jrajahalme any thoughts on this?

jrajahalme replied:
I presume this can happen now due to DNS proxy being initialized earlier in bootstrap, hence this should also be marked as `Fixes: #12964` and backported to 1.7., 1.8, and 1.9.

I'm fine by doing the access log, but leaving out the info we don't have yet. To that end I'd prefer setting the fields not dependent on `secID` and leaving only those out if `secID` is `nil`.

joestringer reviewed (commented):
I wasn't able to to reproduce the issue with an image derived from this change yet, so seems like it is likely to resolve #13881.

aditighag committed:
fqdn: Add a nil check for security id lookup

The security id lookup could return nil if the identity cache
isn't initialized during endpoints restore time, resulting in a crash.
Hence, add a nil check before populating log record values.

Signed-off-by: Aditi Ghag <aditi@cilium.io>
